.github/workflows/approve-dependabot-pr.yml

```diff
@@ -0,0 +1,12 @@
+name: auto approve dependabot pull requests
+on: pull_request
+
+jobs:
+  auto-approve-dependabot-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: auto approve dependabot pull requests
+        uses: hmarr/auto-approve-action@master
+        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+        with:
+          github-token: "${{ secrets.DEPENDABOT_TOKEN }}"
```

Comment:
Can we just use `secrets.GITHUB_TOKEN`? That does not require a custom token to be created... 

https://help.github.com/en/articles/virtual-environments-for-github-actions#github_token-secret

Reply:
I tried using `GITHUB_TOKEN`, but it wouldn't trigger the mergify to auto-merge.

When using `GITHUB_TOKEN`. the pr is approved by `github-action[bot]`, somehow mergify doesn't recognize it as an approved pr so it failed the condition check.

I will test it again, just to confirm.

Reply:
Let's start with opening an issue with mergify and see if they can provide some guidance...

Reply:
Ahh, can you add `github-action[bot]` as a collaborator with write privs to that repo?

Reply:
There were no issues at all. submitted an [issue](https://github.com/Mergifyio/mergify-engine/issues/514) to mergify and got a response instantly.

`
If you are using branch protection, you can safely remove - "#approved-reviews-by>=1". When merge action is run, Mergify always wait the branch protection to be validated before merging the PR.
`
Tested with `GITHUB_TOKEN` and it works. I have updated this pr to reflect the suggestions.